extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlPatch.java

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package io.kubernetes.client.extended.kubectl;
+
+import io.kubernetes.client.common.KubernetesObject;
+import io.kubernetes.client.extended.kubectl.exception.KubectlException;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.util.PatchUtils;
+import io.kubernetes.client.util.PatchUtils.PatchCallFunc;
+
+public class KubectlPatch<ApiType extends KubernetesObject>
+    extends Kubectl.ResourceBuilder<ApiType, KubectlPatch<ApiType>>
+    implements Kubectl.Executable<ApiType> {
+  private String patchFormat;
+  private PatchCallFunc patchCallFunc;
+
+  KubectlPatch(Class<ApiType> apiTypeClass) {
+    super(apiTypeClass);
+  }
+
+  public KubectlPatch patchFormat(String patchFormat) {
+    this.patchFormat = patchFormat;
+    return this;
+  }
+
+  public KubectlPatch patchCallFunc(PatchCallFunc patchCallFunc) {
```

Comment:
this is not necessary, you can did the patch using `GenericKuberneteApi#patch`

Reply:
I don't understand. Do you want me to replace the `PatchUtils` with `GenericKubernetesApi`??

Reply:
i mean `patchCallFunc` is not intuitive, one of the point of our kubectl series here is to make those users already familiar w/ kubectl to get on board our utilities, is it? and yes, replacing `PatchUtils` w/ `GenericKubernetesApi` is exactly what i mean and i expect it can be used like:

```
Kubectl.patch()
   .apiClient(...)
   .resourcePlural(...)
   .namespace("default")
   .name("default")
   .format(xxx')
   .patchContent(..)
   .execute();
```

what do you think?

Reply:
SGTM. I will take a look at this.

Reply:
I have added the above changes.